src/azure-cli/azure/cli/command_modules/iot/shared.py

```diff
@@ -62,3 +62,11 @@ class RenewKeyType(Enum):
     Primary = 'primary'
     Secondary = 'secondary'
     Swap = 'swap'
+
+# pylint: disable=too-few-public-methods
+class AuthenticationType(Enum):
+    """
+    Type of the Authentication for the routing endpoint.
+    """
+    KeyBased = 'keybased'
```

Comment:
nit: Consider removing the "based" suffix from at least the user view. `--auth-type key` or  `--auth-type identity`  still seem to reflect the original intent but with less typing.

Reply:
Actually isn't the official property value in the case of MSI `SystemAssigned` ?

Reply:
```identityBased``` and ```keyBased``` are authentication mechanisms for endpoint communication (storage, servicebus, eventhub, etc) - and ```SystemAssigned``` is for hub identity creation (System Assigned Identity)